c_src/quicer_connection.c

```diff
@@ -465,9 +476,48 @@ async_connect3(ErlNifEnv *env,
 
   enif_monitor_process(NULL, c_ctx, &c_ctx->owner->Pid, &c_ctx->owner_mon);
 
+  ERL_NIF_TERM ecacertfile;
+  X509_STORE *trusted = NULL;
+  if (enif_get_map_value(env, eoptions, ATOM_CACERTFILE, &ecacertfile))
+    {
+      char cacertfile[PATH_MAX];
+      if (enif_get_string(env, ecacertfile, cacertfile,
+                          PATH_MAX, ERL_NIF_LATIN1) > 0)
+        {
+          X509_LOOKUP *lookup = NULL;
+          trusted = X509_STORE_new();
+
+          if (trusted != NULL)
+            {
+              lookup = X509_STORE_add_lookup(trusted, X509_LOOKUP_file());
+              if (lookup != NULL)
+                {
+                  if (!X509_LOOKUP_load_file(lookup, cacertfile,
+                                             X509_FILETYPE_PEM))
+                    {
+                      X509_STORE_free(trusted);
+                      trusted = NULL;
+                    }
+                }
+              else
+                {
+                  X509_STORE_free(trusted);
+                  trusted = NULL;
+                }
+            }
+          c_ctx->trusted = trusted;
+        }
+
+      if (trusted == NULL)
+          fprintf(stderr, "failed to set cacertfile");
```

Comment:
Shall we prefer to return the error instead?

Reply:
Yes, that is probably better.